### DISCUSSION TITLE
is it a bug or a feature?

### DISCUSSION THREAD

**Discussion by Maltiec** [py5coding/py5generator#286] [Q&A] [answered]
A few days ago something unexpected happened: first time in my life I passed negative value as an argument for extent in py5.square - and it worked! So I suppose extent is just added to upper left corner coordinates, and square grows either to downright or to upperleft. I've also tried passing negative value to py5.circle - it worked too, but circle is the same positive or negative. I am curious if anyone use this feature in sketches? ))

**Comment by villares** [comment]
Cheers @Maltiec, I think of it as a feature 😄. But, being more serious, I'm not surprised by it because I used to use Processing Java and Python modes for many years and this has always been the Processing drawing API behaviour.

Note that if you use `rect_mode(CENTER)` your "negative" square will behave like the circle. And if you use `ellipse_mode(CORNER)`, the circle will go up and left as it grows. So to me it feels quite consistent, it is a bit as it would "flip inside out" the geometry based on the origin.

I have used it in animations with `lerp` or `map`, that will "turn stuff inside out". Note also that `arc` works the same.

**Comment by hx2A** [comment, marked as answer]
This sounds like a useful feature. There's no reason why Processing must throw an error for negative values, so why not allow it? This is more forgiving to beginners and might enable someone's creativity.
